samples/tiled-convolution/tiled-convolution.cpp

```diff
@@ -0,0 +1,310 @@
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's ComputeCpp SDK
+ *
+ *  tiled-convolution.cpp
+ *
+ *  Description:
+ *    Sample code that illustrates how to divide data into tiles and lauch
+ *    separate kerenl per tile by using range accessors and parallel for with
+ *    offset in SYCL. See the readme for further information
+ *
+ **************************************************************************/
+#include <SYCL/sycl.hpp>
+#include <chrono>
+#include <ctime>
+#include <iostream>
+
+// configuring the openCL local size
+template <typename index_t>
+struct opencl_configuration_t {
+  static constexpr index_t local_size_m = 16;
+  static constexpr index_t local_size_n = 16;
+};
+
+// define the 2-d matrix size
+template <typename index_t>
+class matrix_size_t {
+ public:
+  const index_t m;
+  const index_t n;
+  matrix_size_t(const index_t m_, const index_t n_) : m(m_), n(n_) {}
+  inline index_t size() const { return (m * n); }
+};
+
+// the tiled based convolution functor
+template <typename read_accessor_t, typename write_accessor_t, typename index_t,
+          typename mat_size_t>
+class conv {
+ private:
+  read_accessor_t fil_acc;      // filter accessor
+  read_accessor_t in_acc;       // input accessor
+  write_accessor_t out_acc;     // output accessor
+  const mat_size_t total_size;  // total input size
+  const mat_size_t fil_size;    // filter size
+
+ public:
+  // constructing the functor
+  conv(read_accessor_t fil_acc_, read_accessor_t in_acc_,
+       write_accessor_t out_acc_, const mat_size_t total_size_,
+       const mat_size_t fil_size_)
+      : fil_acc(fil_acc_),
+        in_acc(in_acc_),
+        out_acc(out_acc_),
+        total_size(total_size_),
+        fil_size(fil_size_) {}
+  void inline operator()(cl::sycl::nd_item<2> item_id) {
+    index_t id_m = item_id.get_global(0);  // global id with offset m
+    index_t id_n = item_id.get_global(1);  // global id with offser n
+    index_t m;                             // input index m
+    index_t fil_m;                         // filter index m
+    index_t n;                             // input index n
+    index_t fil_n;                         // filter index n
+    typename write_accessor_t::value_type val = 0.0;
+    // loop over filter size m and add halo to input m
+    for (fil_m = 0, m = -1; fil_m < fil_size.m; fil_m++, m++) {
+      index_t in_id_m = (id_m + m >= 0) ? id_m + m : 0;
+      in_id_m = (in_id_m < total_size.m) ? in_id_m : total_size.m - 1;
+      // loop over filter size n and add halo to input n
+      for (fil_n = 0, n = -1; fil_n < 3; fil_n++, n++) {
```

Comment:
Is this supposed to be fil_size.n instead of 3? It seems weird that it wouldn't match the previous loop.